services-custom/s3-transfer-manager/src/main/java/software/amazon/awssdk/transfer/s3/S3TransferManager.java

```diff
@@ -54,41 +55,140 @@
+ *             transferManager.downloadDirectory(
+ *                  DownloadDirectoryRequest.builder()
+ *                                          .destination(Paths.get("destination/directory"))
+ *                                          .bucket("bucket")
+ *                                          .listObjectsV2RequestTransformer(l -> l.prefix("prefix"))
+ *                                          .build());
+ *         // Wait for the transfer to complete
+ *         CompletedDirectoryDownload completedDirectoryDownload = directoryDownload.completionFuture().join();
+ *
+ *         // Print out any failed downloads
+ *         completedDirectoryDownload.failedTransfers().forEach(System.out::println);
+ * }
  * <b>Copy an S3 object to a different location in S3</b>
- * {@snippet class = software.amazon.awssdk.transfer.s3.samples.S3TransferManagerSamples region = copy}
+ * {@snippet :
+ *         S3TransferManager transferManager = S3TransferManager.create();
+ *         CopyObjectRequest copyObjectRequest = CopyObjectRequest.builder()
+ *                                                                .sourceBucket("source_bucket")
+ *                                                                .sourceKey("source_key")
+ *                                                                .destinationBucket("dest_bucket")
+ *                                                                .destinationKey("dest_key")
+ *                                                                .build();
+ *         CopyRequest copyRequest = CopyRequest.builder()
+ *                                              .copyObjectRequest(copyObjectRequest)
+ *                                              .build();
+ *
+ *         Copy copy = transferManager.copy(copyRequest);
+ *         // Wait for the transfer to complete
+ *         CompletedCopy completedCopy = copy.completionFuture().join();
+ * }
  */
 @SdkPublicApi
+@ThreadSafe
 @SdkPreviewApi
 public interface S3TransferManager extends SdkAutoCloseable {
 
     /**
-     * Download an object identified by the bucket and key from S3 to a local file. For non-file-based downloads, you may use
+     * Downloads an object identified by the bucket and key from S3 to a local file. For non-file-based downloads, you may use
      * {@link #download(DownloadRequest)} instead.
      * <p>
-     *  The SDK will create a new file if the provided one doesn't exist, otherwise replace the existing file. In the
-     *  event of an error, the SDK will NOT attempt to delete the file, leaving it as-is.
+     * The SDK will create a new file if the provided one doesn't exist. The default permission for the new file depends on
+     * the file system and platform. Users can configure the permission on the file using Java API by themselves.
+     * If the file already exists, the SDK will replace it. In the event of an error, the SDK will <b>NOT</b> attempt to delete
+     * the file, leaving it as-is.
```

Comment:
Can we include a passage on the transfer listeners? Something like

_Users can monitor the progress of the download by attaching a {@link TransferListener}. The provided {@link LoggingTransferListener} logs a basic progress bar; users can also implement their own listeners."_